source/Ship.cpp

```diff
@@ -2738,13 +2760,34 @@ bool Ship::Carry(const shared_ptr<Ship> &ship)
 			ship->SetParent(shared_from_this());
 			ship->isThrusting = false;
 			ship->commands.Clear();
+			
 			// If this fighter collected anything in space, try to store it
 			// (unless this is a player-owned ship).
 			if(!isYours && cargo.Free() && !ship->Cargo().IsEmpty())
 				ship->Cargo().TransferAll(cargo);
-			// Return unused fuel to the carrier, for any launching fighter that needs it.
+			
+			// Return unused fuel and ammunition to the carrier, so they may
+			// be used by the carrier or other fighters.
 			ship->TransferFuel(ship->fuel, this);
 			
+			auto toRestock = map<const Outfit *, int>();
+			for(const auto &hardpoint : ship->Weapons())
+			{
+				const Weapon *weapon = hardpoint.GetOutfit();
+				if(weapon)
+				{
+					const Outfit *ammo = weapon->Ammo();
+					int count = ammo ? ship->OutfitCount(ammo) : 0;
+					if(count > 0)
+						toRestock.emplace(ammo, count);
+				}
+			}
+			for(const auto &stockpile : toRestock)
+			{
+				int unloaded = cargo.Add(stockpile.first, stockpile.second);
```

Comment:
Do we want to re-stock ammo back from fighters to the carriers cargo-space? For a flagship I would consider this undesirable, since the player might want to use the cargo-space for something like plundering instead.

For non-flagships I think it would also be fine to just leave the ammo in the fighter, although I don't have a very strong feeling there either way.

Reply:
One other consideration: in the current implementation of planetary cargo-loading there is no guarantee that ammo returned to the carriers cargo-hold will again be loaded into the carrier on launch.